parcels/grid.py

```diff
@@ -78,7 +79,16 @@ def from_netcdf(cls, filenames, variables, dimensions, **kwargs):
             fields[var] = Field.from_netcdf(var, dimensions, dsets, **kwargs)
         u = fields.pop('U')
         v = fields.pop('V')
-        return cls(u, v, u.depth, u.time, fields=fields)
+
+        # assign time_origin if the time dimension has units and calendar
+        try:
+            time_units = dsets[0][dimensions['time']].units
+            calendar = dsets[0][dimensions['time']].calendar
+            time_origin = num2date(0, time_units, calendar)
```

Comment:
Great, glad to hear it works now! Squashing bugs one at a time ;-)

On Thu, 14 Apr 2016 at 04:37 Joe Scutt Phillips <notifications@github.com>
wrote:

> In parcels/grid.py
> <https://github.com/OceanPARCELS/parcels/pull/59#discussion_r59659677>:
>
> > @@ -78,7 +79,16 @@ def from_netcdf(cls, filenames, variables, dimensions, **kwargs):
> >              fields[var] = Field.from_netcdf(var, dimensions, dsets, **kwargs)
> >          u = fields.pop('U')
> >          v = fields.pop('V')
> > -        return cls(u, v, u.depth, u.time, fields=fields)
> > +
> > +        # assign time_origin if the time dimension has units and calendar
> > +        try:
> > +            time_units = dsets[0][dimensions['time']].units
> > +            calendar = dsets[0][dimensions['time']].calendar
> > +            time_origin = num2date(0, time_units, calendar)
>
> Nice! Looks good, it appears that it makes the calculation for me now when
> time_origin is being assigned in Field. Cool, one less thing to worry
> about! 😎
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/OceanPARCELS/parcels/pull/59/files/cd4818b0ce498a704314881e1389881df26f71a8..581a6324f6a5a2601eb90f1850b62d88c0bb794d#r59659677>
>
-- 
Dr. Erik van Sebille
Lecturer in Oceanography and Climate Change
Grantham Institute & Department of Physics
Imperial College London
London, United Kingdom
---------------------------------
e: E.van-Sebille@imperial.ac.uk
t: +44 7 481 294 686
w: www.erik.vansebille.com